### ISSUE TITLE
Compiler permits creation of nullable array type

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#32169]
In https://github.com/dotnet/roslyn/issues/31991#issuecomment-451384616 it is reported that the compiler permits creation of a nullable array type:

``` c#
#nullable enable
public class C {
    public void M() {
    	_ = new C?(); // error (correct)
        _ = new string[50]?; // no error (incorrect)
        _ = new string[50]?[,]; // no error (incorrect)
    }
}
```

**Comment by gafter** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/32431.
